src/parquet/metadata.h

```diff
@@ -84,6 +85,7 @@ class ApplicationVersion {
 
   // Checks if the Version has the correct statistics for a given column
   bool HasCorrectStatistics(Type::type primitive,
+                            const parquet::format::Statistics& statistics,
```

Comment:
Since this is actually a thrift definition, it is not allowed to be included here apparently. I can move the check for `max_equals_min` out of this but that doesnt seem like the best thing organizationally. Any suggestions here?

Reply:
You may have to create an accessor class to hide the Thrift structures from the public API. That's what we've done with the rest of the Thrift metadata

Reply:
I got around it by not passing the thrift structure as an argument anymore but just whether max/min are equivalent. This seems not ideal since the tests dont cover extracting this value and this logic seems like a good fit for being within `HasCorrectStatistics` so I will probably look into an accessor class. Is there a good example you can point me to (will be faster than me scouring the codebase looking for a good one)?

Reply:
You'll want to use the private implementation ("PIMPL") pattern. Here's an example of one: https://github.com/apache/arrow/blob/master/cpp/src/arrow/memory_pool.h#L108. There are others in this codebase, too